### ISSUE TITLE
Using `apple_framework` to build an external framework fails: `<unknown>:0: error: underlying Objective-C module <module-name> not found`

### ISSUE THREAD

**Issue by jschear** [bazel-ios/rules_ios#307]
When building an external dependency with `apple_framework`, I get the following error:
```
ERROR: /private/var/tmp/_bazel_jschear/7436e6242b3207e4692683d02ddfce6d/external/com_github_apple_swiftcollections/BUILD.bazel:5:16: Compiling Swift module DequeModule failed: (Exit 1): worker failed: error executing command
  (cd /private/var/tmp/_bazel_jschear/7436e6242b3207e4692683d02ddfce6d/execroot/build_bazel_rules_ios && \
  exec env - \
    APPLE_SDK_PLATFORM=iPhoneSimulator \
    APPLE_SDK_VERSION_OVERRIDE=14.5 \
    XCODE_VERSION_OVERRIDE=12.5.1.12E507 \
  bazel-out/host/bin/external/build_bazel_rules_swift/tools/worker/worker swiftc @bazel-out/ios-x86_64-min13.3-applebin_ios-ios_x86_64-dbg-ST-8633e27d3539/bin/external/com_github_apple_swiftcollections/DequeModule.swiftmodule-0.params)
Execution platform: @local_config_platform//:host
<unknown>:0: error: underlying Objective-C module 'DequeModule' not found
<unknown>:0: error: underlying Objective-C module 'DequeModule' not found
<unknown>:0: error: underlying Objective-C module 'DequeModule' not found
<unknown>:0: error: underlying Objective-C module 'DequeModule' not found
<unknown>:0: error: underlying Objective-C module 'DequeModule' not found
<unknown>:0: error: underlying Objective-C module 'DequeModule' not found
<unknown>:0: error: underlying Objective-C module 'DequeModule' not found
<unknown>:0: error: underlying Objective-C module 'DequeModule' not found
<unknown>:0: error: underlying Objective-C module 'DequeModule' not found
<unknown>:0: error: underlying Objective-C module 'DequeModule' not found
<unknown>:0: error: underlying Objective-C module 'DequeModule' not found
<unknown>:0: error: underlying Objective-C module 'DequeModule' not found
<unknown>:0: error: underlying Objective-C module 'DequeModule' not found
<unknown>:0: error: underlying Objective-C module 'DequeModule' not found
<unknown>:0: error: underlying Objective-C module 'DequeModule' not found
<unknown>:0: error: underlying Objective-C module 'DequeModule' not found
<unknown>:0: error: underlying Objective-C module 'DequeModule' not found
<unknown>:0: error: underlying Objective-C module 'DequeModule' not found
<unknown>:0: error: underlying Objective-C module 'DequeModule' not found
<unknown>:0: error: underlying Objective-C module 'DequeModule' not found
<unknown>:0: error: underlying Objective-C module 'DequeModule' not found
<unknown>:0: error: underlying Objective-C module 'DequeModule' not found
```

The build worked before this commit: https://github.com/bazel-ios/rules_ios/commit/c145ef3c1d5c5937195e6d7d40111db2640b197b

Here's the diff between `DequeModule_swift_vfs.yml` on that commit & the commit before it:
```diff
diff --git DequeModule_swift_vfs_old.yaml DequeModule_swift_vfs.yaml
index dc65ecb..96cfd61 100755
--- DequeModule_swift_vfs_old.yaml
+++ DequeModule_swift_vfs.yaml
@@ -1,6 +1,6 @@
 {
   "case-sensitive": true,
-  "overlay-relative": true,
+  "overlay-relative": false,
   "roots":
     [
       {
@@ -16,7 +16,7 @@
                   {
                     "type": "file",
                     "name": "DequeModule-umbrella.h",
-                    "external-contents": "../../../bazel-out/ios-x86_64-min13.3-applebin_ios-ios_x86_64-dbg-ST-8633e27d3539/bin/external/com_github_apple_swiftcollections/DequeModule-modulemap/DequeModule-umbrella.h",
+                    "external-contents": "bazel-out/ios-x86_64-min13.3-applebin_ios-ios_x86_64-dbg-ST-8633e27d3539/bin/external/com_github_apple_swiftcollections/DequeModule-modulemap/DequeModule-umbrella.h",
                   },
                 ],
             },
@@ -28,7 +28,7 @@
                   {
                     "type": "file",
                     "name": "module.modulemap",
-                    "external-contents": "../../../bazel-out/ios-x86_64-min13.3-applebin_ios-ios_x86_64-dbg-ST-8633e27d3539/bin/external/com_github_apple_swiftcollections/DequeModule-modulemap/DequeModule.modulemap",
+                    "external-contents": "bazel-out/ios-x86_64-min13.3-applebin_ios-ios_x86_64-dbg-ST-8633e27d3539/bin/external/com_github_apple_swiftcollections/DequeModule-modulemap/DequeModule.modulemap",
                   },
                 ],
             },
@@ -37,7 +37,7 @@
       {
         "type": "file",
         "name": "/build_bazel_rules_ios/frameworks/DequeModule.swiftmodule",
-        "external-contents": "../../../bazel-out/ios-x86_64-min13.3-applebin_ios-ios_x86_64-dbg-ST-8633e27d3539/bin//DequeModule.swiftmodule",
+        "external-contents": "bazel-out/ios-x86_64-min13.3-applebin_ios-ios_x86_64-dbg-ST-8633e27d3539/bin//DequeModule.swiftmodule",
       },
     ],
   "use-external-names": false,
```

I'll open a PR with a reproducer.

**Comment by jerrymarino** [comment]
@jschear thank you kindly for sending over a repro for this! I think I've got an idea of what's broken here. In the meantime, are you able to use the commit sha before that commit?

**Comment by jschear** [comment]
Yep, that works for us -- thank you for looking into it! 👍 

I was also considering using the HEAD of master and opting out of the VFS feature for now, but I've had trouble figuring out how to do that. If I pass `enable_framework_vfs = False` in an `apple_framework` target, [this condition ends up enabling it anyways.](https://github.com/bazel-ios/rules_ios/blob/a2358ec5a30b756547e0829131e014b3024086a0/rules/library.bzl#L464)

**Comment by jerrymarino** [comment]
You've got it! Ah interesting so you were not using the framework macro before I take it then? e.g. just using `apple_library`?

**Comment by jschear** [comment]
We are using the framework macro, we were just pointing to a commit before the VFS stuff landed 😅

**Comment by tinder-maxwellelliott** [comment]
We are hitting the same issue as well, unable to switch over to virtual frameworks for now

**Comment by jerrymarino** [comment]
Thanks for the heads up! I should have a PR for this here soon, we didn't have this case covered on the CI, and I want to also make sure it works well with PodToBUILD!

**Comment by jerrymarino** [comment]
I think I see the problem of this, it assumes that the build file path is adjacent to the rule.

@tinder-maxwellelliott was this the main blocker to not using virtual frameworks? We had seen a large increase in performance: an order of magnitude better on some swift modules, so hoping to make that the default longer term.

**Comment by tinder-maxwellelliott** [comment]
> I think I see the problem of this, it assumes that the build file path is adjacent to the rule.
> 
> @tinder-maxwellelliott was this the main blocker to not using virtual frameworks? We had seen a large increase in performance: an order of magnitude better on some swift modules, so hoping to make that the default longer term.

We would love to switch over, for our external frameworks they live under `external/Pods/**/BUILD`, would we need to mutate that structure to use VFS?

**Comment by jerrymarino** [comment]
@tinder-maxwellelliott my intention is that it just works out of the box 🤞 !  I have been running with `build --features apple.virtualize_frameworks` in the bazelrc for the past few weeks and it's been mostly great.  

I was hoping since the non-virtual-framework-vfsoverlay is using the same code path ( fixed by #317 ), that it will make it work you. 

If you've got another issue with it, please let me know - it's easiest by a test case for me like in the issues
